system/core/pages.php

```diff
@@ -438,12 +444,19 @@ function wikify($elem) {
 	function wikify_link($node) {
 		global $hyphaXml, $isoLangList;
 
-		$path = explode('/', $node->getAttribute('href'), 3);
-		if (count($path) < 2)
+		$href = $node->getAttribute('href');
+		// This parses a query string of the form en/pagename/subpath#anchor
+		// and splits that into the language and pagename,
+		// followed by an optional subpath and/or anchor. The
+		// language and pagename will be replaced by the page
+		// id, the rest will be kept as-is.
+		//
+		// Note that pipes are used to enclose the regex rather
+		// than slashes, so we can use slashes without escaping
+		// them.
+		if (!preg_match('|^([^/#]+)/([^/#]+)(.*)$|', $href, $matches))
 			return;
-
-		$language = $path[0];
-		$pagename = $path[1];
+		list($all, $language, $pagename, $extra) = $matches;
```

Comment:
Seems like duplicate behaviour to `HyphaRequest`. The method in `HyphaRequest` is way more extensive and seems to cover more cases.